models/arkham/ethereum/arkham_ethereum_airdrop_claims_legacy.sql

```diff
@@ -0,0 +1,46 @@
+{{
+    config(
+        schema = 'arkham_ethereum',
+        alias = alias('airdrop_claims', legacy_model=True),
+        materialized = 'table',
+        file_format = 'delta',
+        tags=['legacy'],
+        unique_key = ['recipient', 'tx_hash', 'evt_index'],
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "arkham",
+                                \'["hildobby"]\') }}'
+    )
+}}
+
+{% set arkm_token_address = '0x6e2a43be0b1d33b726f0ca3b8de60b3482b8b050' %}
+
+WITH early_price AS (
+    SELECT MIN(minute) AS minute
+    , MIN_BY(price, minute) AS price
+    FROM {{ source('prices', 'usd') }}
+    WHERE blockchain = 'ethereum'
+    AND contract_address='{{arkm_token_address}}'
+    )
+
+SELECT 'ethereum' AS blockchain
+, t.evt_block_time AS block_time
+, t.evt_block_number AS block_number
+, 'ApeCoin' AS project
+, 'ApeCoin Airdrop' AS airdrop_identifier
+, t.account AS recipient
+, t.contract_address
+, t.evt_tx_hash AS tx_hash
+, CAST(t.amount AS DECIMAL(38,0)) AS amount_raw
+, CAST(t.amount/POWER(10, 18) AS double) AS amount_original
+, CASE WHEN t.evt_block_time >= (SELECT minute FROM early_price) THEN CAST(pu.price*t.amount/POWER(10, 18) AS double)
+    ELSE CAST((SELECT price FROM early_price)*t.amount/POWER(10, 18) AS double)
+    END AS amount_usd
+, '{{arkm_token_address}}' AS token_address
+, 'APE' AS token_symbol
+, t.evt_index
+FROM {{ source('arkham_ethereum', 'Airdrop_evt_Claimed') }} t
+LEFT JOIN {{ ref('prices_usd_forward_fill_legacy') }} pu ON pu.blockchain = 'ethereum'
+    AND pu.contract_address='{{arkm_token_address}}'
+    AND pu.minute=date_trunc('minute', t.evt_block_time)
+WHERE t.evt_block_time BETWEEN '2022-03-17' AND '2022-06-16'
```

Comment:
how come date filters differ between dunesql and legacy? also, do we still need legacy or is this lineage migrated fully?

Reply:
I just copied the spell from another airdrop, just modifying what's needed. Afaik, dates need to be casted from strings to work in dunesql while just strings are find in spark, hence the different syntax.

Also, I've been adding legacy models to all my spells, it's faster than looking for if it's needed (+ still not too sure when it is/isn't needed tbh)

Reply:
not a syntax difference, but a difference in date range itself. i assume it means you don't care about data in the legacy file?

Reply:
Ahh yeah my bad, there's a bunch I forgot to change on the legacy file to match the DuneSQL one, just fixed those inconsistencies

Reply:
for your awareness, this may be helpful to know when to avoid making legacy changes:
```
comm -23 <( dbt ls -s airdrop_ethereum_claims_legacy+,tag:legacy --resource-type model --output path | sed 's/_legacy.sql/.sql/g'| sort ) <( dbt ls -s airdrop_ethereum_claims+,tag:dunesql --resource-type model --output path | sort )
```

this should output spells which exist on spark, but not yet migrated to dunesql. you can also run the inner commands separate to see the outputs for each `dbt ls`. for example:
![image](https://github.com/duneanalytics/spellbook/assets/102681548/e339382a-eaee-486f-8536-f799ac5d4752)

nothing returns in the overall command, meaning you can safely ignore legacy files since migration is complete (including downstream spells). then you can see each of the inner commands return two spells as expected.

Reply:
you should be able to reuse that, and just change spell names as needed

Reply:
with that in mind, do you want to remove modifications to legacy here? you can also do a follow-up PR if you want to remove the legacy files for full `airdrop_claims` lineage if you want. if not, we'll eventually get to it.

Reply:
Just removed the legacy file @jeff-dude, lmk if there's more I need to do

Reply:
I'm confused, CI tests are asking me to add the legacy file back in, I thought it would be fine without it, what's going on here @jeff-dude ?

<img width="992" alt="Screenshot 2023-10-13 at 12 13 38" src="https://github.com/duneanalytics/spellbook/assets/23564601/5762f623-fcf8-4e1e-aa4f-0f95d8879b8c">

Reply:
grr, okay, we'll add back for now. we're v close to shutting off spark, so we'll be past this soon